### PR TITLE
Emulate remainder for all u64 integers

### PR DESCRIPTION
**Pull request by jonatanklosko**
This aligns the random values between torchx and the binary backend.

### CODE REVIEW

File: torchx/lib/torchx/backend.ex [hunk 1]
```diff
@@ -861,16 +861,62 @@ defmodule Torchx.Backend do
...
-      end
+    {device, _} = left_tx
+
+    if l.type == {:u, 64} do
+      # We emulate u64 numbers with s64.
+
+      # Numbers smaller than max_s64 are kept as positive s64 and fmod
+      # works fine for those.
+
+      reminder_from_positive = Torchx.fmod(left_tx, right_tx)
```
Comment (polvalente) [inline torchx/lib/torchx/backend.ex:872]:
```suggestion
      remainder_from_positive = Torchx.fmod(left_tx, right_tx)
```

File: torchx/lib/torchx/backend.ex [hunk 2]
```diff
@@ -861,16 +861,62 @@ defmodule Torchx.Backend do
...
+      #
+      #   x mod y = ((max_s64 mod y) + (rest mod y)) mod y
+      #
+      # We can safely compute operations on the right hand side using
+      # s64 numbers.
+
+      max_s64_tx =
+        Nx.Constants.max_finite(:s64, backend: Nx.BinaryBackend)
+        |> Nx.to_number()
+        |> Torchx.scalar_tensor(to_torch_type(l.type), device)
```
Comment (polvalente) [inline torchx/lib/torchx/backend.ex:894]: Can't you allocate on Torchx directly with `backend: {Torchx.Backend, device: device}`?

File: torchx/lib/torchx/backend.ex [hunk 3]
```diff
@@ -861,16 +861,62 @@ defmodule Torchx.Backend do
...
+      #
+      #   x = max_s64 + rest
+      #
+      # We can obtain rest as follows:
+      #
+      #   rest = x - max_s64
+      #
+      # We also know that the following holds:
+      #
+      #   x mod y = ((max_s64 mod y) + (rest mod y)) mod y
```
Comment (polvalente) [inline torchx/lib/torchx/backend.ex:886]: nice!

File: torchx/lib/torchx/backend.ex [hunk 4]
```diff
@@ -861,16 +861,62 @@ defmodule Torchx.Backend do
...
+      # s64 numbers.
+
+      max_s64_tx =
+        Nx.Constants.max_finite(:s64, backend: Nx.BinaryBackend)
+        |> Nx.to_number()
+        |> Torchx.scalar_tensor(to_torch_type(l.type), device)
+
+      rest_tx = Torchx.subtract(left_tx, max_s64_tx)
+
+      reminder_from_negative =
```
Comment (polvalente) [inline torchx/lib/torchx/backend.ex]:
```suggestion
      remainder_from_negative =
```

File: torchx/lib/torchx/backend.ex (hunk 2 above)
Comment (jonatanklosko) [reply torchx/lib/torchx/backend.ex:894]: I was hoping for a better way :D

**Review by polvalente — APPROVED** [review]

**Review by josevalim — APPROVED** [review]

**Pull request merged**